src/js/extensions/placeholder.js

```diff
@@ -58,8 +63,14 @@ var Placeholder;
             // Check placeholder on blur
             this.base.subscribe('editableBlur', this.handleBlur.bind(this));
 
+            // if we don't want the placeholder to be removed on click but when user start typing
+            if (this.hideOnClick) {
+                this.base.subscribe('editableClick', this.handleHidePlaceholderEvent.bind(this));
+            } else {
+                this.base.subscribe('editableKeyupDelete', this.handleBlur.bind(this));
```

Comment:
Could you try this:
`this.base.subscribe(`editableKeydown`, this.handleHidePlaceholderEvent.bind(this))`

I think we can avoiding creating all of those new custom events and just use that.  I also think we'll be able to get rid of the listener for `editableKeypress` if we went that route.